tests/gdimageline/github_bug_167.c

```diff
@@ -0,0 +1,73 @@
+/*
+	Testing thick styled horizontal, vertical and diagonal lines.
+
+	We're drawing four triangles with basically the same pattern style. Each
+	triangle has exactly the same shape (right isosceles, with the hypotenuse
+	being diagonal) and size. The triangles are arranged from lower left to
+	upper right with increasing line thickness.
+
+	To have the same pattern style, it has to be adapted according to the
+	thickness (i.e. the pixels have to be repeated thickness times), which
+	happens in setStyleAndThickness().
```

Comment:
i'm lazy and often like to stick a link to the github issue at the end of the comment block too